src/GGTestUtils.py

```diff
@@ -333,10 +471,46 @@ def _upload_files_to_s3(self, files: Sequence[os.PathLike | str],
             print(
                 f"File {file_path} successfully uploaded to {bucket_name}/{object_name}"
             )
+
+        # Wait for S3 artifact propagation by checking availability
+        if files:
+            last_object = object_name    # Use the last uploaded file for checking
+            for attempt in range(20):    # Max 20 seconds
+                try:
+                    self._s3Client.head_object(Bucket=bucket_name,
+                                               Key=last_object)
+                    print(
+                        f"S3 artifact verified accessible after {attempt + 1}s")
+                    break
+                except:
+                    time.sleep(1)
+            else:
+                raise Exception(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.



Similar issue at line number 504.